include/seastar/core/memory.hh

```diff
@@ -139,8 +139,35 @@ static constexpr size_t huge_page_size =
 void configure(std::vector<resource::memory> m, bool mbind,
         std::optional<std::string> hugetlbfs_path = {});
 
+/**
+ * Set the global state of the abort on allocation failure behavior.
+ *
+ * If enabled, an allocation failure (i.e., the requested memory
+ * could not be allocate even after reclaim was attempted), will
+ * generally result in abort() being called. If disabled, the
+ * failure is reported to the caller, e.g., by throwing a
+ * std::bad_alloc for C++ allocations such as new, or returning
+ * a null pointer from malloc.
+ *
+ * Note that even if the global state is set to enabled, the 
+ * disable_abort_on_alloc_failure_temporarily class may override
+ * the behavior tepmorarily on a given shard. That is, abort only
+ * occurs if abort is globablly enabled on this shard and there
+ * are no disable_abort_... objets currently alive on this shard.
+ */
+void set_abort_on_allocation_failure(bool enabled);
+
+/// A deprecated alias for set_abort_on_allocation_failure(true).
+[[deprecated("use set_abort_on_allocation_failure(true) instead")]]
 void enable_abort_on_allocation_failure();
 
+/**
+ * Return true if the global abort on allocation failure behavior
+ * is enabled, or false otherwise. Always returns false if the 
+ * default (system) allocator is being used.
+ */
+bool is_abort_on_allocation_failure();
```

Comment:
Shouldn't this fwd declaration belong to next patch instead?

Reply:
Good catch, moved in next push.